e2e-tests/src/tests/bindings.test.ts

```diff
@@ -0,0 +1,638 @@
+
+    // Returns a 201 otherwise
+    response = await request.post(`${urls.MERLIN_URL}/addExternalDataset`, {
+      data: {
+        action: {name: "addExternalDataset"},
+        input: {
+          planId: plan_id,
+          datasetStart:'2021-001T06:00:00.000',
+          profileSet: {'/my_boolean':{schema:{type:'boolean'},segments:[{duration:3600000000,dynamics:true}],type:'discrete'}},
+          simulationDatasetId: null
+        },
+        request_query: "",
+        session_variables: admin.session}});
+    expect(response.status()).toEqual(201);
+    expect((await response.json()).datasetId).toBeDefined();
+  });
+  test('ExtendExternalDataset', async ({request}) => {
+    // Returns a 404 if the MissionModelId is invalid
+    // message is "no such mission model"
+    let response = await request.post(`${urls.MERLIN_URL}/extendExternalDataset`, {
+      data: {
+        action: {name: "extendExternalDataset"},
+        input: {
+          datasetId:-1,
+          profileSet: {'/my_boolean':{schema:{type:'boolean'},segments:[{duration:3600000000,dynamics:true}],type:'discrete'}}
+        },
+        request_query: "",
+        session_variables: admin.session}});
+    expect(response.status()).toEqual(404);
+    expect((await response.json()).message).toEqual('no such plan dataset');
+
+    // Setup
+    const datasetInput : ExternalDatasetInsertInput = {
+      plan_id: plan_id,
+      profile_set: {'/my_boolean':{schema:{type:'boolean'},segments:[{duration:3600000000,dynamics:true}],type:'discrete'}},
+      dataset_start: '2021-001T06:00:00.000'
+    }
+    const dataset_id = await req.insertExternalDataset(request, datasetInput);
+
+    // Returns a 200 if the ID is valid
+    response = await request.post(`${urls.MERLIN_URL}/extendExternalDataset`, {
+      data: {
+        action: {name: "extendExternalDataset"},
+        input: {
+          datasetId:dataset_id,
+          profileSet: datasetInput.profile_set
+        },
+        request_query: "",
+        session_variables: admin.session}});
+    expect(response.status()).toEqual(200);
+    expect((await response.json())).toEqual({datasetId: dataset_id});
+  });
+  test('ConstraintsDslTypescript', async ({request}) => {
+    // Returns a 200 with a failure status if the MissionModelId is invalid
+    // reason is "No mission model exists with id `-1`"
+    let response = await request.post(`${urls.MERLIN_URL}/constraintsDslTypescript`, {
+      data: {
+        action: {name: "constraintsDslTypescript"},
+        input: {
+          missionModelId:'-1',
+          planId: null,
+        },
+        request_query: "",
+        session_variables: admin.session}});
+    expect(response.status()).toEqual(200);
+    expect((await response.json())).toEqual({ status: 'failure', reason: 'No mission model exists with id `-1`' });
+
+    // Expectation: According to `GenerateConstraintsLibAction::run`, this request should fail
```

Comment:
Why is this test commented out?

Reply:
This is how the test _should_ be if `ConstraintDSLTypescript` behaved as expected. Since it doesn't (see Future Work), it's been left commented out. I can either remove it or add a TODO to uncomment it once the `ConstraintDSLTypescript` endpoint is fixed, if you'd like.

Reply:
Done!